### PR TITLE
added validation to Azure SQL firewall start and end IP address

### PR DESCRIPTION
**Pull request by ritesh-modi**
all acceptance tests passed for resource_arm_sql_firewall_rule

riteshmodi@MININT-3FOKASG terraform-provider-azurerm % make acctests SERVICE='sql' TESTARGS='-run=TestAccAzureRMSqlFirewallRule' TESTTIMEOUT='120m'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./azurerm/internal/services/sql/tests/ -run=TestAccAzureRMSqlFirewallRule -timeout 120m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccAzureRMSqlFirewallRule_basic
=== PAUSE TestAccAzureRMSqlFirewallRule_basic
=== RUN   TestAccAzureRMSqlFirewallRule_requiresImport
=== PAUSE TestAccAzureRMSqlFirewallRule_requiresImport
=== RUN   TestAccAzureRMSqlFirewallRule_disappears
=== PAUSE TestAccAzureRMSqlFirewallRule_disappears
=== CONT  TestAccAzureRMSqlFirewallRule_basic
=== CONT  TestAccAzureRMSqlFirewallRule_disappears
=== CONT  TestAccAzureRMSqlFirewallRule_requiresImport
--- PASS: TestAccAzureRMSqlFirewallRule_disappears (191.29s)
--- PASS: TestAccAzureRMSqlFirewallRule_basic (222.72s)
--- PASS: TestAccAzureRMSqlFirewallRule_requiresImport (222.90s)
PASS
ok  	github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/sql/tests	222.943s

### CODE REVIEW

File: azurerm/internal/services/sql/tests/resource_arm_sql_firewall_rule_test.go [hunk 1]
```diff
@@ -218,13 +233,28 @@ resource "azurerm_resource_group" "test" {
...
   version                      = "12.0"
   administrator_login          = "mradministrator"
   administrator_login_password = "thisIsDog11"
+
+  extended_auditing_policy {
+    storage_endpoint                        = azurerm_storage_account.test.primary_blob_endpoint
+    storage_account_access_key              = azurerm_storage_account.test.primary_access_key
+    storage_account_access_key_is_secondary = true
+    retention_in_days                       = 6
+  }
```
Comment (tombuildsstuff) [inline azurerm/internal/services/sql/tests/resource_arm_sql_firewall_rule_test.go:257]: this shouldn't be needed - unfortunately this is a bug in the Azure API (being tracked in https://github.com/Azure/azure-rest-api-specs/issues/11271) which should be reverted soon, as such could we revert this for the moment?

File: azurerm/internal/services/sql/tests/resource_arm_sql_firewall_rule_test.go [hunk 2]
```diff
@@ -174,13 +174,28 @@ resource "azurerm_resource_group" "test" {
...
```
Comment (tombuildsstuff) [inline azurerm/internal/services/sql/tests/resource_arm_sql_firewall_rule_test.go:198]: (same here) can we remove this - this is an API regression and shouldn't be needed?

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @ritesh-modi 

Thanks for this PR :)

Taking a look through this mostly looks good - if we can revert the extra configuration in the tests (unfortunately there's an API regression which requires specifying this, which we're tracking separately, but can be tracked outside of this PR) then this should otherwise be good to merge 👍 

Thanks!

File: azurerm/internal/services/sql/tests/resource_arm_sql_firewall_rule_test.go (hunk 1 above)
Comment (ritesh-modi) [reply azurerm/internal/services/sql/tests/resource_arm_sql_firewall_rule_test.go]: Sure Tom, removed and fixed it

File: azurerm/internal/services/sql/tests/resource_arm_sql_firewall_rule_test.go (hunk 2 above)
Comment (ritesh-modi) [reply azurerm/internal/services/sql/tests/resource_arm_sql_firewall_rule_test.go]: Sure Tom, removed and fixed it

**Review by tombuildsstuff — APPROVED** [review]
LGTM - thanks for pushing those changes @ritesh-modi  👍

**Comment by ritesh-modi** [comment]
Thanks Tom 👍

**Comment by ritesh-modi** [comment]
Also, I think I re-submitted the build. I am not able to cancel it, please help in cancelling it if not needed. I do not know exactly how the spelling changed from "Desktop Virtualization" to  "DesktopVirtualization". so if this change is not needed, please cancel the newly submitted build.

Thank you for all the help and accepting the PR.
Ritesh

**Pull request merged**